mlx/c/ops.h

```diff
@@ -819,60 +885,96 @@ int mlx_slice_update(
     const int* strides,
     size_t strides_num,
     const mlx_stream s);
-int mlx_softmax(
+int mlx_slice_update_dynamic(
+    mlx_array* res,
+    const mlx_array src,
+    const mlx_array update,
+    const mlx_array start,
+    const int* axes,
+    size_t axes_num,
+    const mlx_stream s);
+int mlx_softmax_axes(
     mlx_array* res,
     const mlx_array a,
     const int* axes,
     size_t axes_num,
     bool precise,
     const mlx_stream s);
-int mlx_softmax_all(
+int mlx_softmax_axis(
+    mlx_array* res,
+    const mlx_array a,
+    int axis,
+    bool precise,
+    const mlx_stream s);
+int mlx_softmax(
     mlx_array* res,
     const mlx_array a,
     bool precise,
     const mlx_stream s);
-int mlx_sort(mlx_array* res, const mlx_array a, int axis, const mlx_stream s);
-int mlx_sort_all(mlx_array* res, const mlx_array a, const mlx_stream s);
-int mlx_split_equal_parts(
+int mlx_sort_axis(
+    mlx_array* res,
+    const mlx_array a,
+    int axis,
+    const mlx_stream s);
+int mlx_sort(mlx_array* res, const mlx_array a, const mlx_stream s);
+int mlx_split(
     mlx_vector_array* res,
     const mlx_array a,
     int num_splits,
     int axis,
     const mlx_stream s);
-int mlx_split(
+int mlx_split_sections_x(
```

Comment:
Does it make sense to name it `mlx_split_sections_axis` or you like the `x` more ? I don't see `x` anywhere else.. so I'm wondering about that.